remill/Arch/X86/Semantics/SSE.cpp

```diff
@@ -1353,6 +1353,88 @@ IF_AVX(DEF_ISEL(VMAXSD_XMMdq_XMMdq_XMMq) = MAXSD<VV128W, V128, V128>;)
 
 namespace {
 
+template <typename D, typename S1, typename S2>
+DEF_SEM(MINPS, D dst, S1 src1, S2 src2) {
+  auto dest_vec = FReadV32(src1);
+  auto src2_vec = FReadV32(src2);
+
+  auto vec_count = NumVectorElems(src2_vec);
+  _Pragma("unroll")
+  for (std::size_t i = 0; i < vec_count; i++) {
+    auto v1 = FExtractV32(dest_vec, i);
+    auto v2 = FExtractV32(src2_vec, i);
+
+    auto min = v1;
+
+    // If either float is a NaN (SNaN or QNaN):
+    if (std::isunordered(v1, v2)) {
+        min = v2;
+    } 
+    // or if both floats are 0.0:
+    else if ((v1 == 0.0) && (v2 == 0.0)) {
```

Comment:
Change style to:

```c++
if (...) {
  min = v2;

// or if both floats are 0.0:
} else if ((v1 == 0.0) && (v2 == 0.0)) {
```

Reply:
Make sure to add positive and negative zero to test cases. I recommend looking into `__builtin_isless` and variants. Possibly just use `CompareFloats` to implement the comparison to choose the minimum value.